crates/backend/src/codegen.rs

```diff
@@ -780,12 +787,43 @@ impl TryToTokens for ast::Export {
         let nargs = self.function.arguments.len() as u32;
         let attrs = &self.function.rust_attrs;
 
-        let start_check = if self.start {
-            quote! { const _ASSERT: fn() = || -> #projection::Abi { loop {} }; }
-        } else {
-            quote! {}
+        let mut checks = Vec::new();
+        if self.start {
+            checks.push(quote! { const _ASSERT: fn() = || -> #projection::Abi { loop {} }; });
         };
 
+        if let Some(class) = self.rust_class.as_ref() {
+            // change span of class, so it points to the location of the
+            // function causing the assert to fail
+            let mut class = class.clone();
+            class.set_span(self.rust_name.span());
```

Comment:
I'm not sure whether this is hygienic (or even correct). I did this so that the error message would point to the function causing the error instead of the type name on the `impl` block. This seems hacky, so please let me know if there are better ways of doing this.